src/byte_offsets.h

```diff
@@ -0,0 +1,62 @@
+#ifndef BYTE_OFFSETS_H
+#define BYTE_OFFSETS_H
+
+#include "redisearch.h"
+#include "varint.h"
+#include "rmalloc.h"
+
+typedef struct __attribute__((packed)) RSByteOffsetMap {
+  // ID this belongs to.
+  uint8_t fieldId;
+
+  // The position of the first token for this field.
+  uint32_t firstTokPos;
+
+  // Position of last token for this field
+  uint32_t lastTokPos;
+} RSByteOffsetField;
+
+typedef struct RSByteOffsets {
+  // By-Byte offsets
+  RSOffsetVector offsets;
+  // List of field-id <-> position mapping
+  RSByteOffsetField *fields;
+  // How many fields
+  uint8_t numFields;
```

Comment:
we support up to 1024 fields, but only up to 32 indexed text fields. So as long as we apply this only to indexed text fields we are good.